### PR TITLE
Check whether semgrep-proprietary still builds

### PR DESCRIPTION
**Pull request by mjambon**
See description in the job config.

Update: this PR is now only about using the script in CI. The script in question is now part of https://github.com/returntocorp/semgrep/pull/6664.

PR checklist:

- [x] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [x] Tests included or PR comment includes a reproducible test plan
- [x] Documentation is up-to-date
- [x] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure on any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

**Comment by mjambon** [comment]
The alpine image we use doesn't come with `ssh` or with `gh`. We can run `apk add openssh` but what about `gh`? It would need to be installed from a tarball as described [here](https://github.com/cli/cli/issues/658#issuecomment-701375745). It's not a huge deal but I'd like to know if there's a way to post a PR comment without installing `gh` into our docker container.

File: scripts/build-semgrep-proprietary [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+$(cd "$semgrep_root" && git status)
+EOF
+
+# Use a folder where it's safe to delete any existing semgrep-proprietary.
+workdir=$semgrep_root/../tmp-semgrep-pro
+mkdir -p "$workdir"
+(
+  cd "$workdir"
+  rm -rf semgrep-proprietary
+  git clone "$semgrep_pro_url" --depth 1
```
Comment (spencerdrak) [inline scripts/build-semgrep-proprietary]: I think auth might need to be configured here, since `semgrep-proprietary` is a private repo.

**Review by spencerdrak — CHANGES_REQUESTED** [review]
Overall, this looks good, but there's three comments I have:
- I think it would be best to wait until after we reelase 1.0.0 to merge this in, so our release process freeze has ended
- The auth (as noted) might need some work. LMK if you need some help with that
- I would strongly suggest testing this in our test GH Actions repository. I am happy to explain how that works, or even better, I can find some time to put together a better set of documents around that. I'll touch base when I get those into Notion.

**Comment by aryx** [comment]
Closing; Someone can reopen when it works.
It just clutters our list of pull requests. We must keep it clean.

**Pull request closed without merging**